app/webserver/handler.go

```diff
@@ -287,3 +270,87 @@ func uploadSendVoiceNote(voiceNote SendVoiceNoteMessage) error {
 	}
 	return nil
 }
+
+func sendProfile(id int64) {
+	recipient := store.RecipientsModel.GetRecipientById(id)
+	if recipient == nil {
+		return
+	}
+	var contact *contacts.Contact
+	if recipient.E164 != "" {
+		contact = store.GetContactForTel(recipient.E164)
+	}
+	profileMessage := &ProfileMessage{
+		Recipient: recipient,
+		Contact:   contact,
+	}
+	profileMessageEnvelope := &ProfileMessageEnvelope{
+		ProfileMessage: profileMessage,
+	}
+	var err error
+	message := &[]byte{}
+	*message, err = json.Marshal(profileMessageEnvelope)
+	if err != nil {
+		log.Errorln("[axolotl-ws] UpdateMessageHandler", err)
+		return
+	}
+	broadcast <- *message
+
+}
+func updateProfileName(id int64, name string) {
+	recipient := store.RecipientsModel.GetRecipientById(id)
+	if recipient == nil {
+		return
+	}
+	recipient.ProfileGivenName = name
+	recipient.SaveRecipient()
+	if recipient.E164 != "" {
+		contactForTel := store.GetContactForTel(recipient.E164)
+		if contactForTel != nil {
+			contactForTel.Name = name
+			contact.EditContact(store.ContactsModel.GetContact(int(id)), *contactForTel)
+		}
+	}
+	sendProfile(id)
+}
+func createChatForRecipient(id int64) {
+	recipient := store.RecipientsModel.GetRecipientById(id)
+	if recipient == nil {
+		return
+	}
+	newChat, err := store.SessionsV2Model.GetOrCreateSessionForDirectMessageRecipient(recipient.Id)
+	if err != nil {
+		log.Errorln("[axolotl] Create chat error:", err)
+	} else {
+
+		activeChat = newChat.ID
+		store.ActiveSessionID = activeChat
+		requestEnterChat(newChat.ID)
+		sendChatList()
+	}
+}
+
+func createRecipient(uuid string) {
+	recipient := store.RecipientsModel.GetOrCreateRecipient(uuid)
+	if recipient == nil {
+		return
+	}
+}
+func createRecipientAndAddToGroup(uuid string, groupId string) {
+	recipient := store.RecipientsModel.GetOrCreateRecipient(uuid)
+	if recipient == nil {
+		return
+	}
+	group, err := store.GroupV2sModel.GetGroupById(groupId)
+	if err != nil {
+		log.Errorln("[axolotl] Create recipient and add to group error:", err)
+		return
+	}
+	err = group.AddMember(recipient)
+	if err != nil {
+		log.Errorln("[axolotl] Create recipient and add to group error:", err)
+		return
+	}
+	requestEnterChat(store.ActiveSessionID)
```

Comment:
When you open a group with a lot of recipients, that haven't been in your recipients db and you click on one of the recipients, this recipient is created and in order to update the group view in axolotl web we reload the group. This is done by `requestEnterChat`. This could be done differently, but then we have to redo some parts in axolotl web.